src/rocker/os_detector.py

```diff
@@ -12,37 +12,25 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import json
 import pexpect
 
-from ast import literal_eval 
 from io import BytesIO as StringIO
 
 from .core import docker_build, get_docker_client
 
 
 DETECTION_TEMPLATE="""
-FROM python:3-slim-stretch as detector
-# Force the older version of debian for detector.
-# GLIBC is forwards compatible but not necessarily backwards compatible for pyinstaller
-# https://github.com/pyinstaller/pyinstaller/wiki/FAQ#gnulinux
-# StaticX is supposed to take care of this but there appears to be an issue when using subprocess
-
-RUN mkdir -p /tmp/distrovenv
-RUN python3 -m venv /tmp/distrovenv
-# patchelf needed for staticx
-# binutils provides objdump needed by pyinstaller
-RUN apt-get update && apt-get install -qy patchelf binutils
-RUN . /tmp/distrovenv/bin/activate && pip install distro pyinstaller==4.0 staticx==0.12.3
-
-RUN echo 'import distro; import sys; output = (distro.name(), distro.version(), distro.codename()); print(output) if distro.name() else sys.exit(1)' > /tmp/distrovenv/detect_os.py
-RUN . /tmp/distrovenv/bin/activate && pyinstaller --onefile /tmp/distrovenv/detect_os.py
-
-RUN . /tmp/distrovenv/bin/activate && staticx /dist/detect_os /dist/detect_os_static && chmod go+xr /dist/detect_os_static
+FROM golang:1.19 as detector
+RUN git clone -q https://github.com/dekobon/distro-detect.git && \
+    cd distro-detect && \
+    git checkout -q 5f5b9c724b9d9a117732d2a4292e6288905734e1 && \
```

Comment:
Could you add a comment about how this has was selected? It looks like it's just the latest commit, but noting that would be good for future reference.